商品詳細表示機能の実装

tomo1107 opened the pull request:
# What
## 実装内容
- 商品詳細表示機能の実装
- 商品購入機能を未実装の為、以下の機能は未実装です。
　1. ログイン状態の出品者でも、売却済みの商品に対しては『商品の編集』『削除』ボタンが表示されないこと
　2. ログイン状態の出品者以外のユーザーでも、売却済みの商品に対しては『購入画面に進む』ボタンが表示されないこと
　3. 売却済みの商品は、画像上に『sold out』の文字が表示されること


## 動作確認結果(Gyazo画像)
- ログイン状態の出品者が、自身の出品した販売中商品の詳細ページへ遷移した動画
https://gyazo.com/6d69a3c0bd5419c5a49bc50dd6f2f171
- ログイン状態の出品者が、自身の出品した売却済み商品の詳細ページへ遷移した動画（現段階で商品購入機能の実装が済んでいる場合）
※商品購入機能を未実装の為、ありません。
- ログイン状態の出品者以外のユーザーが、他者の出品した販売中商品の詳細ページへ遷移した動画
https://gyazo.com/259e5a043ed5c5601278d11cc41bf6cb
- ログイン状態の出品者以外のユーザーが、他者の出品した売却済み商品の詳細ページへ遷移した動画（現段階で商品購入機能の実装が済んでいる場合）
※商品購入機能を未実装の為、ありません。
- ログアウト状態のユーザーが、商品詳細ページへ遷移した動画
https://gyazo.com/5a33daf75e5247a49e12dd66682943ad

# Why
- 商品詳細表示機能が正常に実装出来ているかをご確認お願いいたします。

tomo1107 committed:
商品詳細機能の実装

tomo1107 committed:
Userモデルの単体テストコードの記述方法の変更

Hide-Morishita commented on app/views/items/show.html.erb:
```diff
@@ -16,54 +16,51 @@
-
-
-    <%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
+    <% if current_user == @item.user %>
+      <%= link_to "商品の編集", "#", method: :get, class: "item-red-btn" %>
+      <p class="or-text">or</p>
+      <%= link_to "削除", "#", method: :delete, class:"item-destroy" %>
+    <% elsif user_signed_in? %>
+      <%# 商品が売れていない場合はこちらを表示しましょう %>
+      <%= link_to "購入画面に進む", "#" ,class:"item-red-btn"%>
+      <%# //商品が売れていない場合はこちらを表示しましょう %>
+    <% end %>
```
編集や削除のボタンにも`user_signed_in?`が必要になりますので、
下記を参考にしてみましょう。
コードのイメージは以下のようになります。

```ruby
if ”ログインしていれば” 
  if "現在ログインしているユーザーが商品の出品者であれば"
    商品の編集
    削除
  else
    購入画面に進む
  end
end
```

Hide-Morishita commented on app/views/items/show.html.erb:
```diff
@@ -16,54 +16,51 @@
+          <td class="detail-value"><%= @item.dateOfDelivery.name %></td>
         </tr>
       </tbody>
     </table>
```
```ruby
<a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
```
103行目も修正をお願いいたします。
また
```ruby
<%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
```
こちらは、実装案内用のコメントアウトですので削除しておきましょう。

Hide-Morishita reviewed (commented):
レビュー依頼、キャプチャ動画ありがとうございます。
コメントいたしましたので、ご確認をお願いいたします。

tomo1107 committed:
商品詳細機能の実装(是正1回目)

以下のご指摘に対しての是正対応
https://github.com/tomo1107/furima-35073/pull/5#pullrequestreview-636231003

tomo1107 commented:
> レビュー依頼、キャプチャ動画ありがとうございます。
> コメントいたしましたので、ご確認をお願いいたします。

ご確認ありがとうございました。
ご指摘に対して是正しましたので、再レビューの程、よろしくお願いいたします。

TakashiKozuma commented:
レビュー依頼ありがとうございます。
修正点も含め確認できました。
LGTMです！詳細機能の実装お疲れさまでした。

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に次の実装のためのブランチを切ってから次にお進み下さい。
